Assets/Scripts/Drinkables/Drink Pickup/DrinkChoice.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Drinkables;
+using UnityEngine;
+using UnityEngine.UI;
+using Random = UnityEngine.Random;
+
+public class DrinkChoice : MonoBehaviour
+{
+    [SerializeField] private GameObject menuPanel;
+    [SerializeField] private GameObject buffaloDrink;
+    [SerializeField] private GameObject clearlyDrink;
+    [SerializeField] private GameObject flippedDrink;
+    [SerializeField] private GameObject waterDrink;
+    [SerializeField] private GameObject beerDrink;
+    [SerializeField] private float timeLimit;
+    [SerializeField] private GameObject buffaloButton;
+    [SerializeField] private GameObject clearlyButton;
+    [SerializeField] private GameObject flippedButton;
+
+    private float timeCounter = 0;
+    public bool wasActive;
+
+    void Update()
+    {
+        if (GameObject.Find("PauseCanvas").transform.Find("MainPanel").gameObject.activeSelf && menuPanel.activeSelf)
+        {
+            menuPanel.SetActive(false);
+            wasActive = true;
+        }
+        else if (menuPanel.activeSelf)
+        {
+            wasActive = false;
+            timeCounter += Time.unscaledDeltaTime;
+            transform.Find("Menu").Find("Countdown").gameObject.GetComponent<Text>().text = Math.Round(timeLimit - timeCounter).ToString();
+
+            if (timeCounter > timeLimit)
+            {
+                transform.Find("Menu").Find("Countdown").gameObject.GetComponent<Text>().text = "";
+                LeaveBar();
+            }
+        }
+    }
+
+    private void GenerateRandomDrink()
+    {
+        var randomDrink = Random.Range(0, 3);
+        if (randomDrink == 0)
+        {
+            buffaloButton.SetActive(true);
+        }
+        else if (randomDrink == 1)
+        {
+            clearlyButton.SetActive(true);
+        }
+        else
+        {
+            flippedButton.SetActive(true);
+        }
+    }
+
+    public void ConsumeBuffalo()
+    {
+        LeaveBar();
+        Instantiate(buffaloDrink);
+    }
+
+    public void ConsumeClearlyThere()
+    {
+        LeaveBar();
+        Instantiate(clearlyDrink);
+    }
+
+    public void ConsumeUpsideDown()
+    {
+        LeaveBar();
+        Instantiate(flippedDrink);
+    }
+
+    public void ConsumeWater()
+    {
+        LeaveBar();
+        Instantiate(waterDrink);
+    }
+
+    public void ConsumeBeer()
+    {
+        LeaveBar();
+        Instantiate(beerDrink);
+    }
+
+    public void EnterTheBar()
+    {
+        Cursor.visible = true;
+        Time.timeScale = 0;
```

Comment:
I think that setting the timeScale will break the pause menu, since the pause manu does `1 - Time.timeScale`, so if we enter the bar, the timeScale goes to 0, but if we press pause, then `1 - 0 = 1` so the timeScale in the pause menu is reversed

Reply:
I would introduce a boolean to freeze the movement of all of the characters (ex: `if (inBar) return;`) while in the bar menu to allow the Update's to be called (aka the timer of the bar works fine). This would also allow us to pause everything in the pause menu correctly.

Reply:
We could do that
Or really change the `1 - Time.timeScale` by `menu.selfActive ? 0 : 1`

Reply:
I think that that should work, too